packages/lesswrong/components/posts/ExternalPostImporter.tsx

```diff
@@ -0,0 +1,410 @@
+          }}
+          onChange={(_event: any, editor: any) => {
+            setCommentValue(editor.getData());
+          }}
+        />
+        <div className={classes.editorButtons}>
+          <Button
+            className={classNames(classes.formButton, classes.cancelButton)}
+            onClick={onCancel}
+          >
+            Cancel
+          </Button>
+          <Button
+            type="submit"
+            className={classNames(classes.formButton, classes.submitButton)}
+            onClick={() => onPublish(commentValue)}
+            disabled={isButtonDisabled}
+            title={isButtonDisabled ? 'You must write a review before submitting.' : ''}
+          >
+            Submit Linkpost 
+          </Button>
+        </div>
+      </Components.ContentStyles>
+    </div>
+  );
+};
+
+const ExternalPostImporter = ({ classes }: { classes: ClassesType<typeof styles> }) => {
+  const [value, setValue] = useState('');
+  const [post, setPost] = useState<ExternalPostImportData | null>(null);
+  const [postContent, setPostContent] = useState<string>('');
+  const [published, setPublished] = useState<boolean>(false);
+
+  const { Typography, Loading, ContentStyles } = Components;
+
+  const currentUser = useCurrentUser();
+
+  const [importUrlAsDraftPost, { data, loading, error }] = useLazyQuery(gql`
+    query importUrlAsDraftPost($url: String!) {
+      importUrlAsDraftPost(url: $url) {
+        _id
+        slug
+        title
+        content
+        url
+      }
+    }
+  `);
+
+  const { create } = useCreate({
+    collectionName: 'Comments',
+    fragmentName: 'CommentsList',
+  });
+
+  const { mutate: updatePost } = useUpdate({
+    collectionName: 'Posts',
+    fragmentName: 'PostsList',
+  });
+
+  useEffect(() => {
+    if (data && data.importUrlAsDraftPost) {
+      const importedPost = data.importUrlAsDraftPost;
+      setPost(importedPost);
+      setPostContent(importedPost.content || '');
+    }
+    if (error) {
+      // eslint-disable-next-line no-console
+      console.error(error);
+    }
+  }, [data, error]);
+
+  const handlePublish = async (commentContent: string) => {
+    if (!post || !currentUser) return;
+
+    try {
+      // Create the comment
+      const commentData = {
+        postId: post._id,
+        userId: currentUser._id,
+        reviewingForReview: new Date().getFullYear().toString(),
+        contents: {
+          originalContents: {
+            data: commentContent,
+            type: 'ckEditorMarkup',
+          },
+        } as EditableFieldContents,
+      };
+
+      await create({ data: commentData });
+
+      // Update and publish the post using useUpdate
+      await updatePost({
+        selector: { _id: post._id },
+        data: {
+          contents: {
+            originalContents: {
+              data: postContent,
+              type: 'ckEditorMarkup',
+            },
+          },
+          draft: false,
+          postedAt: new Date(),
```

Comment:
I don't believe this is necessary (this should be getting updated by callbacks on the backend)